### ISSUE TITLE
Zeroize for k256::ecdsa::SigningKey seems broken

### ISSUE THREAD

**Issue by randombit** [RustCrypto/elliptic-curves#691]
With these versions

```
[dependencies]
k256 = "0.11.6"
rand = "0.8"
zeroize = "1.5"
```

this test program fails to compile

```
use k256::ecdsa::SigningKey;
use zeroize::Zeroize;
use rand::thread_rng;

fn main() {
    let mut rng = thread_rng();
    let mut k256 = SigningKey::random(&mut rng);
    k256.zeroize();
}
```

with an error

```
error[E0599]: the method `zeroize` exists for struct `k256::ecdsa::SigningKey`, but its trait bounds were not satisfied
  --> src/main.rs:8:10
   |
8  |     k256.zeroize();
   |          ^^^^^^^ method cannot be called on `k256::ecdsa::SigningKey` due to unsatisfied trait bounds
   |
  ::: /home/jack/.cargo/registry/src/github.com-1ecc6299db9ec823/k256-0.11.6/src/ecdsa/sign.rs:40:1
   |
40 | pub struct SigningKey {
   | ---------------------
   | |
   | doesn't satisfy `k256::ecdsa::SigningKey: DefaultIsZeroes`
   | doesn't satisfy `k256::ecdsa::SigningKey: Zeroize`
   |
   = note: the following trait bounds were not satisfied:
           `k256::ecdsa::SigningKey: DefaultIsZeroes`
           which is required by `k256::ecdsa::SigningKey: Zeroize`

```

is there something I'm missing about how to zeroize this type?

**Comment by tarcieri** [comment]
It zeroizes itself on drop: https://docs.rs/k256/latest/src/k256/ecdsa/sign.rs.html#316-318

It also impls the `ZeroizeOnDrop` trait: https://docs.rs/k256/latest/src/k256/ecdsa/sign.rs.html#316-318

It deliberately *doesn't* impl the `Zeroize` trait because allowing the value to live past zeroization would violate its own invariants (i.e. the secret scalar value is a `NonZeroScalar`)

**Comment by randombit** [comment]
Ah I see. That makes sense. If I can give a bit more context here, I'm trying to wrap a `SigningKey` in another type namely:

```
#[derive(Clone, ZeroizeOnDrop)]
pub struct PrivateKey {
    key: Box<k256::ecdsa::SigningKey>,
}
```

If I don't use `Box` then this works. However I'm trying to `Box` the key because if I don't then the key data is vulnerable to being moved-from and the old stack data will not be zeroized. Am I just out of luck here or is there some other way to handle this?

**Comment by tarcieri** [comment]
@randombit there was a bug in `zeroize_derive`: https://github.com/RustCrypto/utils/pull/787

It seems we merged a PR with a fix but never did a release with that fix. I can go ahead and cut one.

**Comment by tarcieri** [comment]
I just released `zeroize_derive` v1.3.3 which should fix that problem
